creator-node/src/apiSigning.js

```diff
@@ -23,6 +23,33 @@ const generateTimestampAndSignature = (data, privateKey) => {
   return { timestamp, signature: signedResponse.signature }
 }
 
+// Keeps track of cached signatures for `generateTimestampAndSignatureIfNecessary`
+const cachedSignatures = {}
+
+/**
+ * Generates a signature for `data` if only the previous signature
+ * generated is invalid (expired). Otherwise returns an existing signature.
+ * @param {string} data only string data supported
+ * @param {string} privateKey
+ * @returns {string} signature
+ */
+const generateTimestampAndSignatureIfNecessary = (data, privateKey) => {
+  if (data in cachedSignatures) {
+    const signatureTimestamp = cachedSignatures[data].timestamp
+    if (signatureHasExpired(signatureTimestamp)) {
+      // If the signature has expired, remove it from the cache
+      delete cachedSignatures[data]
+    } else {
+      // If the signature has not expired (still valid), use it!
+      return cachedSignatures[data]
+    }
+  }
+  // We don't have a signature already
+  const { timestamp, signature } = generateTimestampAndSignature({ data }, privateKey)
```

Comment:
sorry it's late and maybe a dumb question, but does object-ifying a string input like this even work?

Reply:
it makes { data: 'listens' }. object shorthand